### DISCUSSION TITLE
Propeller blade performance

### DISCUSSION THREAD

**Discussion by Charly-28** [mdolab/dafoam#180] [General]
I would like to optimize the shape of the propeller blade. 
1) My first objective function will be the aerodynamic performance (the ratio of thrust on the torque). How is this function integrated in daFoam?

**Comment by friedenhe** [comment]
We have thrust and torque but do not have their ratio as the objective function yet, you have to implement it.

**Comment by Charly-28** [comment]
Thanks for your reply.
Where should I implement the objective function? 
What are the steps to follow?

**Reply by friedenhe** [reply]
You need to add a child class to DAObjFunc. Refer to src/adjoint/DAObjFunc/DAObjFunForce.C for reference.
